swarm/src/lib.rs

```diff
@@ -676,14 +704,25 @@ where
                     }
                     let peer_id = connected.peer_id;
                     let endpoint = connected.endpoint;
-                    this.behaviour.inject_connection_closed(
-                        &peer_id,
-                        &id,
-                        &endpoint,
-                        handler.into_protocols_handler(),
-                    );
-                    if num_established == 0 {
-                        this.behaviour.inject_disconnected(&peer_id);
+                    let num_established = u32::try_from(established_ids.len()).unwrap();
+                    let conn_was_informed = !this.banned_peer_connections.remove(&id);
+                    if conn_was_informed {
```

Comment:
```suggestion
                    let conn_was_reported = !this.banned_peer_connections.remove(&id);
                    if conn_was_reported {
```

Again, nit pick, sorry, trying to make it easy for future me to understand all of this.

We use the word `reported` above. I would prefer to stay consistent.